dot/types/babe_digest.go

```diff
@@ -17,286 +17,179 @@
 type BabePrimaryPreDigest struct {
-	authorityIndex uint32
-	slotNumber     uint64
-	vrfOutput      [sr25519.VrfOutputLength]byte
-	vrfProof       [sr25519.VrfProofLength]byte
+	AuthorityIndex uint32
+	SlotNumber     uint64
+	VrfOutput      [sr25519.VrfOutputLength]byte
+	VrfProof       [sr25519.VrfProofLength]byte
 }
 
 // NewBabePrimaryPreDigest returns a new BabePrimaryPreDigest
 func NewBabePrimaryPreDigest(authorityIndex uint32, slotNumber uint64, vrfOutput [sr25519.VrfOutputLength]byte, vrfProof [sr25519.VrfProofLength]byte) *BabePrimaryPreDigest {
 	return &BabePrimaryPreDigest{
-		vrfOutput:      vrfOutput,
-		vrfProof:       vrfProof,
-		authorityIndex: authorityIndex,
-		slotNumber:     slotNumber,
+		VrfOutput:      vrfOutput,
+		VrfProof:       vrfProof,
+		AuthorityIndex: authorityIndex,
+		SlotNumber:     slotNumber,
 	}
 }
 
 // ToPreRuntimeDigest returns the BabePrimaryPreDigest as a PreRuntimeDigest
 func (d *BabePrimaryPreDigest) ToPreRuntimeDigest() *PreRuntimeDigest {
-	return NewBABEPreRuntimeDigest(d.Encode())
-}
-
-// Type returns BabePrimaryPreDigestType
-func (d *BabePrimaryPreDigest) Type() byte {
-	return BabePrimaryPreDigestType
-}
-
-// Encode performs SCALE encoding of a BABEPrimaryPreDigest
-func (d *BabePrimaryPreDigest) Encode() []byte {
-	enc := []byte{BabePrimaryPreDigestType}
-
-	buf := make([]byte, 4)
-	binary.LittleEndian.PutUint32(buf, d.authorityIndex)
-	enc = append(enc, buf...)
-
-	buf = make([]byte, 8)
-	binary.LittleEndian.PutUint64(buf, d.slotNumber)
-	enc = append(enc, buf...)
-	enc = append(enc, d.vrfOutput[:]...)
-	enc = append(enc, d.vrfProof[:]...)
-	return enc
-}
-
-// Decode performs SCALE decoding of an encoded BabePrimaryPreDigest, assuming type byte is removed
-func (d *BabePrimaryPreDigest) Decode(r io.Reader) (err error) {
-	d.authorityIndex, err = common.ReadUint32(r)
-	if err != nil {
-		return err
-	}
-
-	d.slotNumber, err = common.ReadUint64(r)
-	if err != nil {
-		return err
-	}
-
-	d.vrfOutput, err = common.Read32Bytes(r)
+	digest := NewBabeDigest()
+	err := digest.Set(*d)
 	if err != nil {
-		return err
+		return nil
 	}
-
-	d.vrfProof, err = common.Read64Bytes(r)
+	enc, err := scale.Marshal(digest)
 	if err != nil {
-		return err
+		return nil
 	}
-	return nil
+	return NewBABEPreRuntimeDigest(enc)
 }
 
-// AuthorityIndex returns the digest's authority index
-func (d *BabePrimaryPreDigest) AuthorityIndex() uint32 {
-	return d.authorityIndex
+// Index Returns VDT index
+func (d BabePrimaryPreDigest) Index() uint { return 1 }
+
+// GetAuthorityIndex returns the digest's authority index
+func (d BabePrimaryPreDigest) GetAuthorityIndex() uint32 {
+	return d.AuthorityIndex
 }
 
-// SlotNumber returns the digest's slot number
-func (d *BabePrimaryPreDigest) SlotNumber() uint64 {
-	return d.slotNumber
+// GetSlotNumber returns the digest's slot number
+func (d BabePrimaryPreDigest) GetSlotNumber() uint64 {
```

Comment:
why did you add the `Get` prefix on these methods?

Reply:
When I made the attributes public I was getting naming errors since the function was the same name as the variable. I just put get to differentiate the two, but can change to something more appropriate. Do you have any suggestion on how to handle this?